hashery/DSOHashery.m

```diff
@@ -0,0 +1,199 @@
+//
+//  DSOHashery.m
+//  hashery
+//
+//  Created by Tong Xiang on 5/20/15.
+//  Copyright (c) 2015 DoSomething.org. All rights reserved.
+//
+
+#import "DSOHashery.h"
+
+/**
+ * Convert integers to string representation.
+ *
+ * Caveats / Limitations
+ *  - the number of words to use in a code is dictated by the number of arrays
+ * provided. If three word arrays are provided, the code will be three words long.
+ *  - word arrays provided must be the same length 
+ *  - the maximum number of digits able to be represented: [array count] ^ (number of arrays)
+ *
+ */
+
+@implementation DSOHashery
+
+- (BOOL)setWordArrayLists:(NSArray *)wordArrayLists
+{
+    if (!wordArrayLists || !wordArrayLists.count) {
+        NSAssert(false, @"setWordArrayLists must be passed a valid array of arrays containing hash words.");
+    }
+    
+    NSInteger testSize = 0;
+    
+    for (int i = 0; i < wordArrayLists.count; i++) {
+        NSArray *arrayIndex = [wordArrayLists objectAtIndex:i];
+        if (i == 0) {
+            testSize = arrayIndex.count;
+            if (testSize == 0) {
+                NSAssert(false, @"setWordArrayLists must be passed a valid array of arrays containing hash words.");
+            }
+        }
+        else if (testSize != arrayIndex.count) {
+            NSAssert(false, @"setWordArrayLists must be passed an array of arrays of equal length.");
+        }
+    }
+    
+    _words = wordArrayLists;
+    _codeLength = _words.count;
+    
+    _base = [[_words objectAtIndex:0] count];
+    
+    NSInteger base = _base;
+    for (int i = 1; i < _words.count; i++) {
+        base *= _base;
+    }
+    _maxIntegerValueOfHash = base - 1;
+    NSLog(@"_maxIntegerValueOfHash: %zd", _maxIntegerValueOfHash);
+    return true;
```

Comment:
Best practice is to use `YES` and `NO` for Booleans: https://github.com/DoSomething/ios-RBReviewer/issues/47